Add troubleshooting page (#533)
napari/docs: docs/troubleshooting.md

# Troubleshooting

This page collects a few known issues and solutions for common problems that users might encounter when using napari.
If you are facing an error that is not covered by this page, please

> [open an issue](https://github.com/napari/napari/issues/new) in the napari repository.

Reviewer: Maybe "check open issues to see if any are related, and if nothing is immediately obvious, open a new issue at..."?

## Known issues

### Resetting preferences and settings

> Changing napari versions—or conda envs with different napari versions—can cause issues with settings. These issues can

Reviewer 2: ```suggestion
Changing napari versions—updating or downgrading—within an environment can cause issues with settings. 
These issues can manifest as napari failing to start. 
Likewise, issues with napari on multiple monitors or with window sizes are also due to settings. 
These issues can frequently
```
I think the envs thing is fixed, there should be no cross talk anymore.

be solved by resetting the settings back to default using the terminal command:

```python
napari --reset
```

### torch + rosetta + napari crashes Python

In some cases, when using napari on Mac M1 using Rosetta you may experience crashing when adding layers or using a script.

> If you observe this, you can try to run napari using the native arm64 Python interpreter (see [napari#7259](https://github.com/napari/napari/issues/7259).

Reviewer: ```suggestion
If you observe this, you should run napari using the native arm64 Python interpreter (see [napari#7259](https://github.com/napari/napari/issues/7259).
```

### "module napari has no attribute Viewer"

This may happen when doing an editable install on top of a normal one, or when you have a local file called `napari.py`.

To fix this, either rename the file or remove all previous napari versions in your environment before installing the editable version.

### My image renders as all black/all white

> This can happen when the contrast limits are not set correctly. You can reset the contrast limits by right-clicking "contras limits", then clicking the "Reset" button in the dialog shown.

Reviewer: ```suggestion
This can happen when the contrast limits are not set correctly. You can reset the contrast limits by right-clicking "contrast limits", then clicking the "Reset" button in the dialog shown.
```

But I think the advice should be more extensive here:

> The contrast limits are the minimum and maximum values displayed by the layer. Any value under the smaller contrast limit will appear as black, while any value above the larger limit will appear as white. Additionally, when all the values are *near* the bottom of the range, the image can still appear black even if there are some very dark but not quite black pixels there.
> 
> You can reset the contrast limits by clicking the Auto-Contrast: "Once" button, or, if that fails, right clicking on the contrast limits to do manual adjustment, including directly editing the numbers shown. When in doubt, you can hover over the canvas with your image layer selected, and check the status message as you move the mouse around. That will give you an idea of the range and variability of the data in your layer, helping you set meaningful contrast limits.

This maybe points to writing a more extensive contrast limits and colormaps tutorial that we can link to from here, but otherwise I think it's worth doing a couple of paragraphs to help people understand what they are reading. (Screenshots would also be good here I think.)

Author: Maybe this is content for the viewer tutorial? Or do you think we need a dedicated page for this?

Author: Or here: https://napari.org/stable/howtos/layers/image.html#adjusting-contrast-limits

Reviewer 2: Yeah, I think we need better documentation for contrast limits and then we can keep this part brief. see also https://github.com/napari/docs/issues/535

Author: Ok - I created #542 and updated this PR to link to that doc (when it's live).

### napari + numba combination doesn't work on admin installs of Windows

If you have trouble using napari together with numba on Windows, try setting `NUMBA_CACHE_DIR` to an user-accessible location.

> See [napari#7288](https://github.com/napari/napari/issues/7288).

Reviewer: I think this should be worded as "I get a PermissionError when trying to launch napari on Windows", since it's not at all clear from the error message that numba has anything to do with it.

## FAQ

### How do I use napari in a remote Jupyter env?

> napari is a native Qt app so it's not possible to do this directly within a Jupyter notebook or JupyterLab tab. However, we are working on documenting options for this workflow.

Reviewer: I would remove this until we actually have the documentation to link to
